### PR TITLE
Codespace

### PR DESCRIPTION
**Pull request by olzhas**
Adding appropriate configuration files to enable github codespace
***

### CODE REVIEW

**Comment by olzhas** [comment]
I am not sure how opengl render might work in this setting.

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,82 @@
+FROM mcr.microsoft.com/devcontainers/cpp:0-ubuntu-20.04
+
+# [Optional] Uncomment this section to install additional vcpkg ports.
+# RUN su vscode -c "${VCPKG_ROOT}/vcpkg install <your-port-name-here>"
+
```
Comment (jslee02) [inline .devcontainer/Dockerfile:5]: Hm, why do we need vcpkg? I thought it's for Windows

```suggestion
```

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,82 @@
+FROM mcr.microsoft.com/devcontainers/cpp:0-ubuntu-20.04
```
Comment (jslee02) [inline .devcontainer/Dockerfile:1]: Why not use `ubuntu:20.04`?

**Review by jslee02 — COMMENTED** [review]
Thank you for doing this!

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (jslee02) [reply .devcontainer/Dockerfile:1]: Actually, we have a [`Dockerfile`](https://github.com/dartsim/dart-ci-docker-containers/blob/master/v6.13/Dockerfile.ubuntu.focal) that is used for CI and prebuilt image (`dartsim/dart-dev:ubuntu-focal-v6.13`). Maybe we could use it instead without installing any additional packages below? The image is hosted in https://hub.docker.com/repository/docker/dartsim/dart-dev

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -1,82 +1,74 @@
-FROM mcr.microsoft.com/devcontainers/cpp:0-ubuntu-20.04
+FROM dartsim/dart-dev:ubuntu-focal-v6.13
 
-# [Optional] Uncomment this section to install additional vcpkg ports.
-# RUN su vscode -c "${VCPKG_ROOT}/vcpkg install <your-port-name-here>"
+# RUN apt-get update && \
```
Comment (jslee02) [inline .devcontainer/Dockerfile:3]: It seems we can remove all the lines below. Other than that, looks good to me!

**Comment by olzhas** [comment]
@jslee02 what do you think about reusing https://github.com/dartsim/dart/blob/main/Dockerfile ?

**Comment by jslee02** [comment]
> @jslee02 what do you think about reusing https://github.com/dartsim/dart/blob/main/Dockerfile ? I discovered that it is already in the repo.

That image is supposed to be an image that already have DART installed in it (so the users don't care about building and installing it but just work in the container). It actually could use `dart-dev` as base image like the Dockerfile in this PR.

I'm merging this PR. Thanks!

**Pull request merged**